src/ConfigAggregator.php

```diff
@@ -92,6 +96,34 @@ public function getMergedConfig()
         return $this->config;
     }
 
+    /**
+     * Throw an exception if duplicate config providers are found.
+     *
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     * @throws InvalidConfigProviderException
+     */
+    private function removeDuplicateProviders(iterable $providers): iterable
+    {
+        $uniqueProviders = [];
+
+        foreach ($providers as $provider) {
+            if (in_array($provider, $uniqueProviders)) {
+                if (is_object($provider)) {
+                    $provider = $provider::class;
```

Comment:
[This comment](https://github.com/laminas/laminas-config-aggregator/issues/36#issuecomment-1710510150) only raised the question of two classes being of the same type (not same instance).

If I understand correctly, you say that we should accept this:
```php
$config = new ConfigAggregator([
    Foo::class, // introduces service interface XY
    Bar::class, // overrides service interface XY with whatever
    Foo::class, // overrides service interface XY with whatever back to the initial service
]);
```
as long as the two occurrences of `Foo` are not the same instance?

Reply:
I am on mobile, just quick look, is array_unique() with SORT_REGULAR constant should already cover this? ref https://3v4l.org/gDm7T

Reply:
My initial plan was to use `array_unique`, but because different types of providers (arrays, class-strings, closures, generators) needed different treatment in the code, I opted for the more complicated solution.
Now, if we agree that this feature should only focus on class-strings, than I can simplify things.

Reply:
I thought of array_unique too but duplicate entries would need to be found and listed for this to have a useful message.

Reply:
> If I understand correctly, you say that we should accept this:

No, that is exactly the case that can be safely detected and reported.


This, however, would be requiring a comparison of internal state which is a no go, really. This should be accepted:

```php
$config = new ConfigAggregator([
    new Foo(), // introduces service interface XY
    Bar::class, // overrides service interface XY with whatever
    new Foo(), // overrides service interface XY with whatever back to the initial service
]);
```